__tests__/schema/thread/create-comment.ts

```diff
@@ -23,79 +23,69 @@ import { gql } from 'apollo-server'
+import { DiscriminatorType } from '~/model/decoder'
 import { encodeThreadId } from '~/schema/thread/utils'
 
-test('unauthenticated user gets error', async () => {
-  mockEndpointsForThreads(article, [[comment1]])
-
-  await assertFailingGraphQLMutation({
-    mutation: gql`
-      mutation ($input: ThreadCreateCommentInput!) {
-        thread {
-          createComment(input: $input) {
-            success
+const mutation = new Client({ userId: user.id }).prepareQuery({
+  query: gql`
+    mutation ($input: ThreadCreateCommentInput!) {
+      thread {
+        createComment(input: $input) {
+          success
+          record {
+            archived
+            content
+            id
           }
         }
       }
-    `,
-    variables: {
-      input: {
-        content: 'Hello',
-        threadId: encodeThreadId(comment1.id),
-        subscribe: true,
-        sendEmail: false,
-      },
+    }
+  `,
+  variables: {
+    input: {
+      content: 'Hello',
+      threadId: encodeThreadId(comment1.id),
+      subscribe: true,
+      sendEmail: false,
     },
-    client: createTestClient({ userId: null }),
-    expectedError: 'UNAUTHENTICATED',
-  })
+  },
 })
 
-test('comment gets created, cache mutated as expected', async () => {
-  const client = createTestClient({ userId: user.id })
-  global.server.use(createUuidHandler(user))
+beforeEach(() => {
+  givenThreads({ uuid: article, threads: [[comment1]] })
+})
 
-  global.server.use(
-    createMessageHandler({
-      message: {
-        type: 'ThreadCreateCommentMutation',
-        payload: {
-          userId: user.id,
-          content: 'Hello',
-          threadId: comment1.id,
-          subscribe: true,
-          sendEmail: false,
-        },
-      },
-      body: {
-        __typename: 'Comment',
-        id: nextUuid(comment1.id),
-        trashed: false,
-        alias: `/mathe/${nextUuid(comment1.id)}/`,
-        authorId: user.id,
-        title: null,
-        date: '2014-03-01T20:45:56Z',
-        archived: false,
-        content: 'Hello',
-        parentId: comment1.id,
-        childrenIds: [],
-      },
+test('comment gets created, cache mutated as expected', async () => {
+  given('UuidQuery').for(user)
+  given('ThreadCreateCommentMutation')
+    .withPayload({
+      userId: user.id,
+      content: 'Hello',
+      threadId: comment1.id,
+      subscribe: true,
+      sendEmail: false,
+    })
+    .returns({
+      __typename: DiscriminatorType.Comment,
+      id: nextUuid(comment1.id),
+      trashed: false,
+      alias: castToAlias(`/mathe/${nextUuid(comment1.id)}/`),
+      authorId: user.id,
+      title: null,
+      date: '2014-03-01T20:45:56Z',
+      archived: false,
+      content: 'Hello',
+      parentId: comment1.id,
+      childrenIds: [],
     })
-  )
-
-  mockEndpointsForThreads(article, [[comment1]])
 
-  //fill cache
-  await client.executeOperation({
+  const query_comments = new Client().prepareQuery({
```

Comment:
```suggestion
  const commentsQuery = new Client().prepareQuery({
```
To maintain standard (camel case and type at end)